lib/global_stats.rb

```diff
@@ -1,13 +1,95 @@
 class GlobalStats
   include StatsHelpers
 
+  GOVUK_ROOT_CONTENT_ID = "f3bbdec2-0e62-4520-a7fd-6ffd5d36e03a".freeze
+
+  BLACKLIST_DOCUMENT_TYPES = %w[
+    redirect
+    staff_update
+    coming_soon
+    travel_advice
+    html_publication
+    manual_section
+    hmrc_manual_section
+    contact
+    completed_transaction
+    service_standard_report
+    employment_tribunal_decision
+    tax_tribunal_decision
+    utaac_decision
+    dfid_research_output
+    asylum_support_decision
+    employment_appeal_tribunal_decision
+    cma_case
+    need
+    working_group
+    organisation
+    person
+    worldwide_organisation
+    world_location
+    topical_event
+    policy_area
+    field_of_operation
+    ministerial_role
+    topical_event_about_page
+    finder_email_signup
+    mainstream_browse_page
+    topic
+    homepage
+    licence_finder
+    search
+    taxon
+    travel_advice_index
+    business_support_finder
+    finder
+    about
+    about_our_services
+    personal_information_charter
+    equality_and_diversity
+    our_governance
+    services_and_information
+    our_energy_use
+    corporate_report
+    social_media_use
+    access_and_opening
+    membership
+    publication_scheme
+    media_enquiries
+    complaints_procedure
+    help_page
+    service_manual_homepage
+    service_manual_service_toolkit
+    service_manual_service_standard
+    service_manual_guide
+    service_manual_topic
+    gone
+  ].freeze
+
   def run
-    total_items = HTTP.get_json("https://www.gov.uk/api/search.json?count=0&debug=include_withdrawn")
-    gauge "items", total_items.fetch("total")
+    total_items = Services.rummager.search(
+      count: 0,
+      debug: 'include_withdrawn'
+    ).to_h.fetch("total")
+
+    taggable_items = Services.rummager.search(
+      count: 0,
+      reject_content_store_document_type: BLACKLIST_DOCUMENT_TYPES,
+      debug: 'include_withdrawn'
+    ).to_h.fetch("total")
+
+    root_taxons = Services.publishing_api.get_links(GOVUK_ROOT_CONTENT_ID)
```

Comment:
So here we return all root level taxons, including draft taxons and the legacy taxonomies right? What are we aiming to filter out by doing this? ie what were we getting before that wasn't desirable?

Reply:
This `get_links` request will only return the root level taxons: the 26 branches or "root taxons" as shown in [content-tagger](https://content-tagger.publishing.service.gov.uk/branches). This includes both draft and published taxons, but only those 26 taxons.

The filter applied here is subtly different to the previous implementation because we're returning content that is only tagged to the Topic Taxonomy using the [`part_of_taxonomy_tree`](https://github.com/alphagov/govuk-tagging-monitor/pull/31/files#diff-f848932934982b2910aff905b7dffdc0R85) attribute with those root taxons, instead of _any_ taxonomy. 

Perhaps at the moment this change isn't drastically different to the previous implementation as there is only the one taxonomy, but in the future, not all taxons will necessarily be associated with the Topic Taxonomy - there could be other taxonomies that exist (at least that's what I've understood from the [Trello card](https://trello.com/c/LXQwobRv/173-investigate-improving-the-itemswithtaxons-metric-to-not-include-taxons-which-are-not-yet-part-of-the-single-taxonomy)...)

Reply:
> Perhaps at the moment this change isn't drastically different to the previous implementation as there is only the one taxonomy, but in the future, not all taxons will necessarily be associated with the Topic Taxonomy - there could be other taxonomies that exist (at least that's what I've understood from the Trello card...)

There is nothing wrong with this reasoning, but, to add another motivation, the imported branches are currently still listed as root taxons, but it would be good to remove them, and for the metrics to only reflect the taxons reachable from the homepage.

So, just creating a taxon and tagging some content to it ideally wouldn't mean that its counted as tagged to the taxonomy, that taxon has to be connected on as well (which is what this change does).

Reply:
Ah ok, that makes sense. I think it might be good to add something about this to the commit message, for easier future perusal...

Reply:
I've added some more detail in the commit https://github.com/alphagov/govuk-tagging-monitor/pull/31/commits/57172a16f10c31dcd4d42ef81e94a0f5bde6a08b

```
Include content only covered by single sitewide taxonomy

In the previous search request, we surface any content tagged to _any_
taxon. Potentially, these taxons might not yet be tagged to the Topic
Taxonomy and would incorrectly skew the content coverage metric.

Now the search is filtered to root taxons and their ancestors to
accurately count the content that is tagged to the Topic Taxonomy by
using the part_of_taxonomy_tree field.
```